scripts/zkai_attention_derived_d128_projection_boundary_gate.py

```
@@ -0,0 +1,943 @@
+            raise AttentionDerivedD128ProjectionBoundaryError("mutation inventory drift")
+        if data.get("case_count") != len(EXPECTED_MUTATIONS):
+            raise AttentionDerivedD128ProjectionBoundaryError("case count drift")
+        if data.get("all_mutations_rejected") is not True:
+            raise AttentionDerivedD128ProjectionBoundaryError("not all mutations rejected")
+        expected_cases = run_mutation_cases(build_core_payload(context), context)
+        expected_outcomes = [(case["name"], case["accepted"], case["rejected"]) for case in expected_cases]
+        outcomes = []
+        for index, (expected_name, case_value) in enumerate(zip(EXPECTED_MUTATIONS, cases, strict=True)):
+            case = _dict(case_value, f"case {index}")
+            if set(case) != {"name", "accepted", "rejected", "error"}:
+                raise AttentionDerivedD128ProjectionBoundaryError("mutation case field drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)